server/src/modules/user/user.service.ts

```diff
@@ -102,6 +107,41 @@ export class UserService {
     }
   }
 
+  async getUserById(id: string): Promise<User> {
+    try {
+      return await this.userModel.findById(id, {
+        _id: 0,
+        role: 0,
+        password: 0,
+      });
+    } catch (err) {
+      throw new BadRequestException('User not found');
+    }
+  }
+
+  async updateUserDetails(id: string, body: UpdateUserDataRequestBodyDto) {
+    try {
```

Comment:
why are we adding a try-catch block in the service file? adding this comment for the second time!